301-sonarqube-azuresql/azuredeploy.json

```diff
@@ -0,0 +1,399 @@
+      "type": "Microsoft.Network/networkSecurityGroups",
+      "location": "[resourceGroup().location]",
+      "apiVersion": "2015-06-15",
+      "dependsOn": [ ],
+      "tags": {
+        "displayName": "SQnsgApp"
+      },
+      "properties": {
+        "securityRules": [
+          {
+            "name": "Allow_RDP_In",
+            "properties": {
+              "description": "Allow RDP In",
+              "protocol": "Tcp",
+              "sourcePortRange": "*",
+              "destinationPortRange": "3389",
+              "sourceAddressPrefix": "Internet",
+              "destinationAddressPrefix": "*",
+              "access": "Allow",
+              "priority": 100,
+              "direction": "Inbound"
+            }
+          },
+          {
+            "name": "Allow_HTTPS_In",
+            "properties": {
+              "description": "Allow HTTPS In",
+              "protocol": "Tcp",
+              "sourcePortRange": "*",
+              "destinationPortRange": "443",
+              "sourceAddressPrefix": "Internet",
+              "destinationAddressPrefix": "*",
+              "access": "Allow",
+              "priority": 101,
+              "direction": "Inbound"
+            }
+          },
+          {
+            "name": "Allow_SonarQube_In",
+            "properties": {
+              "description": "Allow SonarQube In",
+              "protocol": "Tcp",
+              "sourcePortRange": "*",
+              "destinationPortRange": "9000",
+              "sourceAddressPrefix": "Internet",
+              "destinationAddressPrefix": "*",
+              "access": "Allow",
+              "priority": 102,
+              "direction": "Inbound"
+            }
+          },
+          {
+            "name": "Allow_RDP_Out",
+            "properties": {
+              "description": "Allow RDP Out",
+              "protocol": "Tcp",
+              "sourcePortRange": "*",
+              "destinationPortRange": "3389",
+              "sourceAddressPrefix": "[variables('SQvnetExternalSubnetPrefix')]",
+              "destinationAddressPrefix": "[variables('SQvnetInternalSubnetPrefix')]",
+              "access": "Allow",
+              "priority": 100,
+              "direction": "Outbound"
+            }
+          },
+          {
+            "name": "Block_All_In",
+            "properties": {
+              "description": "Block everything else",
+              "protocol": "*",
+              "sourcePortRange": "*",
+              "destinationPortRange": "*",
+              "sourceAddressPrefix": "Internet",
+              "destinationAddressPrefix": "*",
+              "access": "Deny",
+              "priority": 200,
+              "direction": "Inbound"
+            }
+          }
+        ]
+      }
+    }
+  ],
+  "outputs": {
+    "url": {
+      "type": "string",
+      "value": "[concat('http://', parameters('sq_PublicIP_DnsPrefix'), '.' ,resourceGroup().location, '.', 'cloudapp.azure.com:9000') ]"
```

Comment:
this could break if we ever change the scheme for constructing FQDNs (which has happened before). Consider using the reference function like this: "value": "[concat('http://', reference(resourceId('Microsoft.Network/publicIPAddresses', variables('publicIPAddressName'))).dnsSettings.fqdn, '/')]",